### PR TITLE
Stringtask

### PR DESCRIPTION
**Pull request by rila19**
1. Version / Upload

### CODE REVIEW

File: String/Aufgabe3MonthAsString.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            }
+        }
+        DateFormatSymbols a = new DateFormatSymbols();
+        int cool = Integer.parseInt(monatZahl);
+        String[] b = a.getMonths();
+        String c = b[cool-1];
+        return(c);
+    }
+
+    public static void main(String[] args) {
```
Comment (msalauyou) [inline String/Aufgabe3MonthAsString.java:39]: Could you please test it with such dates: 
"03.10.2023"
"03.11.2023"
"03.12.2023"

File: String/Aufgabe4Anagram.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    Ein Anagramm ist ein Satz, der durch Umstellen aller Buchstaben eines anderen
+    Satzes gebildet werden kann. Gross- und Kleinschreibung wird dabei ignoriert.
+    Ebenso spielen Leerzeichen keine Rolle. Aus dem Wort "Desperation" läßt sich zum
+    Beispiel der Satz "A Rope Ends It" bilden.
+    Implementieren eine statische Methode isAnagram, mit der zwei Zeichenketten
+    daraufhin überprüft werden können, ob sie ein Anagram bilden.
+    Verwenden Sie die folgende Signatur für die Methode:
+    public static boolean isAnagram(String str)
+     */
+    public static boolean isAnagram(){
```
Comment (msalauyou) [inline String/Aufgabe4Anagram.java]: Method should accept 2 Strings, in your case it doesn't accept anything. Please re-implement your code to accept 2 strings and do a check.

File: String/Aufgabe5Palindrom.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Implementieren Sie eine statische Methode palindrom, die aus einem Text alle
+    Palindrome auf der Konsole ausgibt. Folgender Aufruf
+    palindrom("Der vor dir stehende Rentner heißt Otto Uwe") sollte z.B. folgendes
+    auf der Konsole ausgeben:
+    Rentner
+    Otto
+    Verwenden Sie die folgende Signatur für die Methode:
+    public static void palindrom(String text)
+     */
+    public static  void splitStringInArray(){
```
Comment (msalauyou) [inline String/Aufgabe5Palindrom.java]: The name of the method doesn't reflect the action it does. I would name it as `printPolyndroms` or something like that. Also it should accept a String, a sentence which it parses and prints polyndromes.

File: String/AufgabeE.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        Gegeben sei folgendes Snippet:
+                ...
+                String result = "";
+                for(int i = 0; i < 10000; i++) {
+                    result += generateRandomString()
+                }
+                ...
+        Optimieren Sie diesen Code.
+    */
+    public static String Sprint9_Aufgabe(){
```
Comment (msalauyou) [inline String/AufgabeE.java]: please rename the method, should start from lower case, should follow camel case and be a verb that reflects what method does.

**Review by msalauyou — CHANGES_REQUESTED** [review]
@rila19, I am very satisfied with your PR. Code looks good. Please see some of my comments. And I don't see the answers on theoretical questions, I remember you had it in your read me file, please add it as well.

**Review by msalauyou — APPROVED** [review]
@rila19 nice work! Thank you!
The code have some format issues, please don't forget to formatt it by pressing Ctrl + Alt + L :)

File: String/Aufgabe3MonthAsString.java (hunk 1 above)
Comment (msalauyou) [reply String/Aufgabe3MonthAsString.java:39]: Fixed

File: String/Aufgabe4Anagram.java (hunk 2 above)
Comment (msalauyou) [reply String/Aufgabe4Anagram.java]: Fixed

File: String/Aufgabe5Palindrom.java (hunk 3 above)
Comment (msalauyou) [reply String/Aufgabe5Palindrom.java]: Fixed

File: String/AufgabeE.java (hunk 4 above)
Comment (msalauyou) [reply String/AufgabeE.java]: Fixed

**Comment by rila19** [comment]
Merge PR

**Comment by rila19** [comment]
Close

**Pull request closed without merging**